docs generation is now failing:

```

File "/home/docs/checkouts/readthedocs.org/user_builds/pandera/envs/2313/lib/python3.11/site-packages/sphinx/ext/autosummary/__init__.py", line 947, in process_generate_options
--
  | generate_autosummary_docs(
  | File "/home/docs/checkouts/readthedocs.org/user_builds/pandera/envs/2313/lib/python3.11/site-packages/sphinx/ext/autosummary/generate.py", line 627, in generate_autosummary_docs
  | name, obj, parent, modname = import_ivar_by_name(entry.name)
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | File "/home/docs/checkouts/readthedocs.org/user_builds/pandera/envs/2313/lib/python3.11/site-packages/sphinx/ext/autosummary/__init__.py", line 817, in import_ivar_by_name
  | real_name, obj, parent, modname = import_by_name(name, prefixes)
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | File "/home/docs/checkouts/readthedocs.org/user_builds/pandera/envs/2313/lib/python3.11/site-packages/sphinx/ext/autosummary/__init__.py", line 759, in import_by_name
  | raise ImportExceptionGroup('no module named %s' % ' or '.join(tried), exceptions)
  | sphinx.ext.autosummary.ImportExceptionGroup: no module named pandera.api.polars.container
  |  
  | The above exception was the direct cause of the following exception:
  |  
  | Traceback (most recent call last):
  | File "/home/docs/checkouts/readthedocs.org/user_builds/pandera/envs/2313/lib/python3.11/site-packages/sphinx/cmd/build.py", line 414, in build_main
  | app = Sphinx(
  | ^^^^^^^
  | File "/home/docs/checkouts/readthedocs.org/user_builds/pandera/envs/2313/lib/python3.11/site-packages/sphinx/application.py", line 332, in __init__
  | self._init_builder()
  | File "/home/docs/checkouts/readthedocs.org/user_builds/pandera/envs/2313/lib/python3.11/site-packages/sphinx/application.py", line 414, in _init_builder
  | self.events.emit('builder-inited')
  | File "/home/docs/checkouts/readthedocs.org/user_builds/pandera/envs/2313/lib/python3.11/site-packages/sphinx/events.py", line 415, in emit
  | raise ExtensionError(
  | sphinx.errors.ExtensionError: Handler <function process_generate_options at 0x7d90459b7ec0> for event 'builder-inited' threw an exception (exception: no module named pandera.api.polars.container)
```

--- a/pandera/api/polars/container.py
+++ b/pandera/api/polars/container.py
@@ -1,9 +1,11 @@
 """DataFrame Schema for Polars."""
 
+from __future__ import annotations
+
 import os
 import sys
 import warnings
-from typing import overload
+from typing import TYPE_CHECKING, overload
 
 from pandera.api.dataframe.container import DataFrameSchema as _DataFrameSchema
 from pandera.api.polars.types import PolarsCheckObjects, PolarsFrame
@@ -16,6 +18,9 @@ if sys.version_info < (3, 11):
     from typing_extensions import Self
 else:
     from typing import Self
+
+if TYPE_CHECKING:  # pragma: no cover
+    import polars as pl
 
 
 class DataFrameSchema(_DataFrameSchema[PolarsCheckObjects]):
@@ -73,6 +78,13 @@ class DataFrameSchema(_DataFrameSchema[PolarsCheckObjects]):
             )
 
         return output
+
+    @overload  # type: ignore[override]
+    def coerce_dtype(self, check_obj: pl.LazyFrame) -> pl.LazyFrame: ...
+    @overload
+    def coerce_dtype(self, check_obj: pl.DataFrame) -> pl.DataFrame: ...
+    def coerce_dtype(self, check_obj):
+        return super().coerce_dtype(check_obj)
 
     @_DataFrameSchema.dtype.setter  # type: ignore[attr-defined]
     def dtype(self, value) -> None:

--- a/pandera/api/polars/model.py
+++ b/pandera/api/polars/model.py
@@ -244,6 +244,15 @@ class DataFrameModel(_DataFrameModel[pl.LazyFrame, DataFrameSchema]):
     def empty(cls: type[Self], *_args) -> DataFrame[Self]:
         """Create an empty DataFrame with the schema of this model."""
         schema = copy.deepcopy(cls.to_schema())
-        schema.coerce = True
-        empty_df = schema.coerce_dtype(pl.DataFrame(schema=[*schema.columns]))
-        return DataFrame[Self](cast(pl.DataFrame, empty_df))
+        if schema.columns:
+            schema.coerce = True
+            empty_df = (
+                schema.coerce_dtype(pl.DataFrame(schema=[*schema.columns]))
+                .lazy()
+                .collect()
+            )
+        elif isinstance(schema.dtype, pe.PydanticModel):
+            empty_df = pl.DataFrame(schema=schema.dtype._get_polars_schema())
+        else:
+            empty_df = pl.DataFrame()
+        return DataFrame[Self](empty_df)

--- a/pandera/engines/polars_engine.py
+++ b/pandera/engines/polars_engine.py
@@ -6,6 +6,7 @@ import datetime
 import decimal
 import inspect
 import logging
+import types
 import warnings
 from collections.abc import Iterable, Mapping, Sequence
 from typing import (
@@ -901,6 +902,32 @@ class PydanticModel(DataType):
         if PYDANTIC_V2:
             return list(self.type.model_fields.keys())  # type: ignore[attr-defined]
         return list(self.type.__fields__.keys())  # type: ignore[attr-defined]
+
+    def _get_polars_schema(self) -> dict[str, DataTypeClass]:
+        """Derive a {col_name: polars_dtype} mapping from the Pydantic model."""
+        from typing import Union, get_args, get_origin
+
+        annotations: dict[str, Any] = {}
+        if hasattr(self.type, "model_fields"):
+            for name, info in self.type.model_fields.items():
+                annotations[name] = info.annotation
+        else:
+            # TODO: remove pydantic v1 branch after dropping v1 support
+            for name, info in getattr(self.type, "__fields__").items():
+                annotations[name] = getattr(info, "outer_type_")
+
+        schema: dict[str, DataTypeClass] = {}
+        for name, ann in annotations.items():
+            origin = get_origin(ann)
+            if origin is Union or isinstance(ann, types.UnionType):
+                non_none = [a for a in get_args(ann) if a is not type(None)]
+                if len(non_none) == 1:
+                    ann = non_none[0]
+            try:
+                schema[name] = convert_py_dtype_to_polars_dtype(ann)
+            except (TypeError, ValueError):
+                schema[name] = pl.Null
+        return schema
 
     def _check_column_names(
         self,

--- a/tests/polars/test_polars_container.py
+++ b/tests/polars/test_polars_container.py
@@ -146,6 +146,22 @@ def test_basic_polars_lazyframe_check_error(
     assert validated_df.equals(ldf_basic.collect())
 
 
+def test_coerce_dtype_preserves_lazyframe(ldf_basic, ldf_schema_basic):
+    """coerce_dtype called with a LazyFrame returns a LazyFrame at runtime."""
+    ldf_schema_basic._coerce = True
+    result = ldf_schema_basic.coerce_dtype(ldf_basic)
+    assert isinstance(result, pl.LazyFrame)
+
+
+def test_coerce_dtype_with_dataframe(ldf_schema_basic):
+    """coerce_dtype is callable with a DataFrame input via the overload path."""
+    ldf_schema_basic._coerce = True
+    df = pl.DataFrame(schema=[*ldf_schema_basic.columns])
+    result = ldf_schema_basic.coerce_dtype(df)
+    # Backend currently lazifies DataFrame inputs; assert it's still a frame-like.
+    assert isinstance(result, (pl.DataFrame, pl.LazyFrame))
+
+
 @pytest.mark.xfail(
     condition=narwhals_installed,
     reason="Narwhals backend does not support dtype coercion",

--- a/tests/polars/test_polars_model.py
+++ b/tests/polars/test_polars_model.py
@@ -82,6 +82,17 @@ def test_empty() -> None:
     assert Schema.validate(df).is_empty()  # type: ignore [attr-defined]
 
 
+def test_empty_no_columns() -> None:
+    """Test empty() on a DataFrameModel with no field annotations."""
+
+    class EmptySchema(DataFrameModel):
+        pass
+
+    df = EmptySchema.empty()
+    assert isinstance(df, pl.DataFrame)
+    assert df.shape == (0, 0)
+
+
 @pytest.fixture
 def ldf_model_with_custom_column_checks():
     class ModelWithCustomColumnChecks(DataFrameModel):

--- a/tests/polars/test_polars_pydantic_dtype.py
+++ b/tests/polars/test_polars_pydantic_dtype.py
@@ -125,6 +125,86 @@ def test_pydantic_model_empty_dataframe_missing_column():
         PydanticSchema.validate(df)
 
 
+def test_pydantic_model_empty_via_model():
+    """Test that DataFrameModel.empty() works with PydanticModel dtype."""
+
+    class Row(BaseModel):
+        name: str | None = None
+        value: int | None = None
+
+    class MySchema(pa.DataFrameModel):
+        class Config:
+            dtype = PydanticModel(Row)
+            coerce = True
+            strict = False
+
+    result = MySchema.empty()
+    assert isinstance(result, pl.DataFrame)
+    assert result.shape[0] == 0
+    assert result.schema == {"name": pl.String, "value": pl.Int64}
+
+
+def test_pydantic_model_get_polars_schema_unsupported_type():
+    """Test that _get_polars_schema falls back to pl.Null for unsupported types."""
+
+    class Custom:
+        pass
+
+    class ModelWithCustom(BaseModel):
+        class Config:
+            arbitrary_types_allowed = True
+
+        field: Custom
+
+    pm = PydanticModel(ModelWithCustom)
+    schema = pm._get_polars_schema()
+    assert schema == {"field": pl.Null}
+
+
+def test_pydantic_model_get_polars_schema_typing_union():
+    """typing.Union[X, None] (not PEP 604) should hit the `origin is Union` branch."""
+    import typing
+    from types import SimpleNamespace
+
+    optional_int = typing.Union[int, None]
+    fake_field = SimpleNamespace(outer_type_=optional_int)
+    fake_model = type("FakeModel", (), {"__fields__": {"a": fake_field}})
+
+    pm = PydanticModel.__new__(PydanticModel)
+    object.__setattr__(pm, "type", fake_model)
+
+    assert pm._get_polars_schema() == {"a": pl.Int64}
+
+
+def test_pydantic_model_get_polars_schema_multiarg_union():
+    """Union with >1 non-None arg can't be unwrapped and falls back to pl.Null."""
+    import typing
+    from types import SimpleNamespace
+
+    multi_union = typing.Union[int, str, None]
+    fake_field = SimpleNamespace(outer_type_=multi_union)
+    fake_model = type("FakeModel", (), {"__fields__": {"x": fake_field}})
+
+    pm = PydanticModel.__new__(PydanticModel)
+    object.__setattr__(pm, "type", fake_model)
+
+    assert pm._get_polars_schema() == {"x": pl.Null}
+
+
+def test_pydantic_model_get_polars_schema_v1_fallback():
+    """Test _get_polars_schema v1 branch via a mock without model_fields."""
+    from types import SimpleNamespace
+
+    fake_field = SimpleNamespace(outer_type_=str | None)
+    fake_model = type("FakeModel", (), {"__fields__": {"name": fake_field}})
+
+    pm = PydanticModel.__new__(PydanticModel)
+    object.__setattr__(pm, "type", fake_model)
+
+    schema = pm._get_polars_schema()
+    assert schema == {"name": pl.String}
+
+
 @pytest.mark.parametrize("coerce", [True, False])
 def test_pydantic_model_coerce_always_true(coerce: bool):
     """Test that DataFrameSchema.coerce is always True with PydanticModel."""

